libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/CiMitUtilityService.java

```diff
@@ -27,24 +31,38 @@ public boolean isBreachingCiThreshold(ContraIndicators contraIndicators) {
     }
 
     public boolean isBreachingCiThresholdIfMitigated(ContraIndicator ci, ContraIndicators cis) {
+        Map<String, ContraIndicatorConfig> contraIndicatorConfigMap =
+                configService.getContraIndicatorConfigMap();
         var scoreOnceMitigated =
-                cis.getContraIndicatorScore(configService.getContraIndicatorConfigMap())
-                        + configService
-                                .getContraIndicatorConfigMap()
-                                .get(ci.getCode())
-                                .getCheckedScore();
+                cis.getContraIndicatorScore(contraIndicatorConfigMap)
+                        + contraIndicatorConfigMap.get(ci.getCode()).getCheckedScore();
         return scoreOnceMitigated
                 > Integer.parseInt(configService.getSsmParameter(CI_SCORING_THRESHOLD));
     }
 
     public Optional<JourneyResponse> getCiMitigationJourneyStep(ContraIndicators contraIndicators)
             throws ConfigException, MitigationRouteConfigNotFoundException {
-        // Try to mitigate an unmitigated ci to resolve the threshold breach
         var cimitConfig = configService.getCimitConfig();
-        for (var ci : contraIndicators.getContraIndicatorsMap().values()) {
+        // Check for multiple unmitigated ci's
+        Collection<ContraIndicator> contraIndicatorsMapValues =
+                contraIndicators.getContraIndicatorsMap().values();
+        if (checkIfMultipleCIForMitigation(contraIndicatorsMapValues)) {
+            return Optional.of(new JourneyResponse(JOURNEY_MULTIPLE_UNMITIGATED_CIS_PATH));
+        }
```

Comment:
Just for separation I'd exposed new event which also take us to no-match page (as done by fail-with-ci)  in case of multiple ci's for mitigation.
As discussed will revert changes done as part of multiple similar CI's fix and will pick master changes.